server/admin/src/main/java/org/kaaproject/kaa/server/admin/services/KaaAdminServiceImpl.java

```diff
@@ -1555,6 +1556,12 @@ public TopicDto editTopic(TopicDto topic) throws KaaAdminServiceException {
             if (isEmpty(topic.getId())) {
                 topic.setCreatedUsername(getCurrentUser().getUsername());
                 checkApplicationId(topic.getApplicationId());
+                for (DataStruct topicDataStruct : clientProvider.getClient().getTopicByAppId(topic.getApplicationId())) {
```

Comment:
This should be more efficient. If there is no method in DAO to get topic by appId and name - add one.